src/source/geojson_source.js

```diff
@@ -110,6 +111,7 @@ class GeoJSONSource extends Evented implements Source {
 
         if (options.maxzoom !== undefined) this.maxzoom = options.maxzoom;
         if (options.type) this.type = options.type;
+        if (options.attribution) this.attribution = options.attribution;
```

Comment:
The attribution option is optional so is that needed? If the attribution option isn't provided, or if it's an empty string it won't set any attribution on the source and the AttributionControl won't add any attribution for it.

Reply:
sorry I was unclear– I was thinking of validating that it is a string (as opposed to another type) _and_ that its "non-empty." I was thinking about just white space, not a true nonempty which would be truthy). the first part is already happening in the style validation code which I forgot, and I'm not sure its worth worrying about the white space case after all because we don't do that for any other style spec values. sorry for confusion!

Reply:
That's right the style validation already spews out an error for a non-string value (handy if someone uses something like `true` by mistake.

As for white space, I'd say we allow it, if someone wants an attribution of "" or "    " we should just comply.